rust/arrow/src/datatypes/interval.rs

```diff
@@ -0,0 +1,33 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use super::{ArrowNumericType, ArrowPrimitiveType, DataType, IntervalUnit};
+use crate::{make_numeric_type, make_type};
+
+make_type!(
+    IntervalYearMonthType,
+    i32,
```

Comment:
fwiw, I do not think this is correct and I opened a [issue for this](https://issues.apache.org/jira/browse/ARROW-11645). I think that this should be `struct(i32,i32)` with C representation or something like that.

As it stands, we will be trying to read an `i32` offseted by 4 bytes from an `i64`, which is asking us to write `unsafe` in a lot of places.

These numbers are not something that suits Rust or anything; they _**must**_ be set according to the Arrow specification.

So, I think that we should hold of this PR until we have the interval implementation correct. In general, I am against adding anything to DataFusion until we have integration tests for the type.

Reply:
@jorgecarleitao You are right, it's will be better.